src/Microsoft.AspNetCore.Sockets/HttpConnectionDispatcher.cs

```diff
@@ -338,31 +338,12 @@ private ConnectionState CreateConnection(HttpContext context)
                 buffer = stream.ToArray();
             }
 
-            IList<Message> messages;
-            if (string.Equals(context.Request.ContentType, MessageFormatter.TextContentType, StringComparison.OrdinalIgnoreCase))
-            {
-                var reader = new BytesReader(buffer);
-                messages = ParseSendBatch(ref reader, MessageFormat.Text);
-            }
-            else if (string.Equals(context.Request.ContentType, MessageFormatter.BinaryContentType, StringComparison.OrdinalIgnoreCase))
-            {
-                var reader = new BytesReader(buffer);
-                messages = ParseSendBatch(ref reader, MessageFormat.Binary);
-            }
-            else
-            {
-                // Legacy, single message raw format
-
-                var format =
-                    string.Equals(context.Request.Query["format"], "binary", StringComparison.OrdinalIgnoreCase)
-                        ? MessageType.Binary
-                        : MessageType.Text;
-                messages = new List<Message>()
-                {
-                    new Message(buffer, format, endOfMessage: true)
-                };
-            }
-
+            var reader = new BytesReader(buffer);
+            var messageFormat =
+                string.Equals(context.Request.ContentType, MessageFormatter.TextContentType, StringComparison.OrdinalIgnoreCase)
```

Comment:
We should return `400 Bad Request` if the content type doesn't match either type.